File: 0169-majority-element/0169-majority-element.cpp
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        my_map = {}
        
        for num in nums :
            if num in my_map :
                my_map[num] += 1
            else :
                my_map[num] = 1
        ans = None
        count = float('-inf')
        
        for key,value in my_map.items() :
            if value > count :
                ans = key
                count = value
        
        return ans

File: 0169-majority-element/0169-majority-element.java
import java.util.Map.Entry;
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> myMap = new HashMap<>();
        int len = nums.length;
        for(Integer num : nums){
            if(myMap.containsKey(num)){
                myMap.put(num,myMap.get(num) + 1);
            }
            else{
                myMap.put(num,1);
            }
        }
        for(Entry<Integer,Integer> entry : myMap.entrySet()){
            if(entry.getValue() > len/2) return entry.getKey();
        }
        return 0;
    }
}